Stacks panics with 'Failed to deserialize vm.Value: Error("missing field `pre_comments`")'

**Describe the bug**
I've built the `develop` (https://github.com/stacks-network/stacks-blockchain/commit/2eea67546c39e54ff2bb74d8a538367615e033e6) branch to run on a mainnet node, and it crashes with the following output:
```
INFO [1695059816.193324] [testnet/stacks-node/src/main.rs:82] [main] stacks-node 2.4.0.0.0 (wallet:a5f3f23302a2f8c76e1bc99f84e1e18c86f7b395, release build, linux [x86_64])
INFO [1695059816.193357] [testnet/stacks-node/src/main.rs:140] [main] Loading config at path /src/stacks-node/Config.toml
INFO [1695059816.276221] [testnet/stacks-node/src/event_dispatcher.rs:1004] [main] Registering event observer at: localhost:3700
INFO [1695059816.285628] [stackslib/src/chainstate/coordinator/mod.rs:3199] [main] Migrating sortition DB to the latest schema version
INFO [1695059816.286634] [stackslib/src/chainstate/coordinator/mod.rs:3203] [main] Migrating chainstate DB to the latest schema version
INFO [1695059816.289297] [testnet/stacks-node/src/run_loop/neon.rs:404] [main] Start syncing Bitcoin headers, feel free to grab a cup of coffee, this can take a while
INFO [1695059816.609092] [stackslib/src/burnchains/burnchain.rs:1495] [main] Syncing Bitcoin blocks: 100.0% (808315 to 808316 out of 808318)
INFO [1695059816.845536] [testnet/stacks-node/src/run_loop/neon.rs:342] [main] Miner node: checking UTXOs at address: 1Bzr6gS4zjMpDDvtPNYbonxjPbTeJPHfsW
INFO [1695059816.846736] [testnet/stacks-node/src/run_loop/neon.rs:347] [main] UTXOs found - will run as a Miner node
INFO [1695059818.947959] [testnet/stacks-node/src/neon_node.rs:3906] [main] Will bootstrap from peers [02da7a464ac770ae8337a343670778b93410f2f3fef6bea98dd1c3e9224459d36b@18000009+00000001://35.194.79.91:20444, 02afeae522aab5f8c99a00ddf75fbcb4a641e052dd48836408d9cf437344b63516@18000009+00000001://35.194.79.91:20444, 03652212ea76be0ed4cd83a25c06e57819993029a7b9999f7d63c36340b34a4e62@18000009+00000001://35.194.79.91:20444]
INFO [1695059818.979495] [testnet/stacks-node/src/neon_node.rs:4318] [main] Start HTTP server on: 0.0.0.0:20443
INFO [1695059818.979509] [testnet/stacks-node/src/neon_node.rs:4319] [main] Start P2P server on: 0.0.0.0:20444
INFO [1695059819.513025] [stackslib/src/chainstate/burn/db/sortdb.rs:1498] [chains-coordinator-0.0.0.0:20443] PoX recipient chosen, recipient: 14-b67ba85d58fbe16155fef693eed529047571ef17, block_height: 800688
INFO [1695059819.513328] [stackslib/src/chainstate/burn/db/sortdb.rs:1498] [chains-coordinator-0.0.0.0:20443] PoX recipient chosen, recipient: 16-9982ebecae89be1c5a8a03d95e134c4a1704a5f3, block_height: 800688
INFO [1695059819.694382] [stackslib/src/chainstate/burn/db/sortdb.rs:4832] [chains-coordinator-0.0.0.0:20443] ACCEPTED(800688) leader block commit 7e4296a3ba4bcbde747e69568d0c4254acbdec572776fcaa8025c743474d1b24 at 800688,99, apparent_sender: 1Kiq1dyVBzYLWGrBPWjChvKyzB2H95x5RJ
INFO [1695059819.695486] [stackslib/src/chainstate/burn/db/sortdb.rs:4832] [chains-coordinator-0.0.0.0:20443] ACCEPTED(800688) leader block commit 325aa90e26277c49412ce7cc9a14770e8bf498d531d10bb7718c50e23133a7e4 at 800688,100, apparent_sender: 18hofVCQv8cMDpqDNqSXyrUDffq7MKLxiu
INFO [1695059819.696494] [stackslib/src/chainstate/burn/db/sortdb.rs:4832] [chains-coordinator-0.0.0.0:20443] ACCEPTED(800688) leader block commit b1f00f75f3640c53917f00bececd4d663e4ac00b90bdf8c58adfa0cab5a79a4b at 800688,101, apparent_sender: 1UiW61F9xnBnnC1J69aPJhFm3ZXvZEkCm
WARN [1695059819.795323] [stackslib/src/net/p2p.rs:2711] [p2p-(0.0.0.0:20444,0.0.0.0:20443)] local.1://(bind=0.0.0.0:20444)(pub=None): failed to learn public IP: NoSuchNeighbor
INFO [1695059819.835453] [stackslib/src/net/http.rs:1687] [p2p-(0.0.0.0:20444,0.0.0.0:20443)] Handle HTTPRequest, verb: GET, peer_addr: 127.0.0.1:36164, path: /v2/info, query:
WARN [1695059819.835845] [stackslib/src/net/p2p.rs:2711] [p2p-(0.0.0.0:20444,0.0.0.0:20443)] local.1://(bind=0.0.0.0:20444)(pub=None): failed to learn public IP: NoSuchNeighbor
ERRO [1695059819.908940] [testnet/stacks-node/src/main.rs:58] [chains-coordinator-0.0.0.0:20443] Process abort due to thread panic: panicked at 'Failed to deserialize vm.Value: Error("missing field `pre_comments`", line: 1, column: 468)', clarity/./src/vm/database/structures.rs:124:1
INFO [1695059819.937705] [stackslib/src/net/p2p.rs:2593] [p2p-(0.0.0.0:20444,0.0.0.0:20443)] IP address changed from None to Some((00000000000000000000ffff0a000e10, 20444))
ERRO [1695059820.152544] [testnet/stacks-node/src/main.rs:60] [chains-coordinator-0.0.0.0:20443] Panic backtrace:    0: stacks_node::main::{{closure}}
             at src/testnet/stacks-node/src/main.rs:59:18
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/boxed.rs:2007:9
      std::panicking::rust_panic_with_hook
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:709:13
   2: std::panicking::begin_panic_handler::{{closure}}
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:597:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/sys_common/backtrace.rs:151:18
   4: rust_begin_unwind
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:593:5
   5: core::panicking::panic_fmt
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:67:14
   6: core::result::unwrap_failed
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/result.rs:1651:5
   7: core::result::Result<T,E>::expect
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/result.rs:1033:23
      <clarity::vm::contracts::Contract as clarity::vm::database::structures::ClarityDeserializable<clarity::vm::contracts::Contract>>::deserialize
             at src/clarity/./src/vm/database/structures.rs:67:17
   8: clarity::vm::database::clarity_db::ClarityDatabase::fetch_metadata::{{closure}}::{{closure}}
             at src/clarity/./src/vm/database/clarity_db.rs:632:40
      core::option::Option<T>::map
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/option.rs:1075:29
      clarity::vm::database::clarity_db::ClarityDatabase::fetch_metadata::{{closure}}
             at src/clarity/./src/vm/database/clarity_db.rs:632:32
      core::result::Result<T,E>::map
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/result.rs:746:25
      clarity::vm::database::clarity_db::ClarityDatabase::fetch_metadata
             at src/clarity/./src/vm/database/clarity_db.rs:630:9
      clarity::vm::database::clarity_db::ClarityDatabase::get_contract
             at src/clarity/./src/vm/database/clarity_db.rs:718:39
   9: clarity::vm::costs::TrackerData::load_costs
             at src/clarity/./src/vm/costs/mod.rs:758:46
  10: clarity::vm::costs::LimitedCostTracker::new
             at src/clarity/./src/vm/costs/mod.rs:665:9
  11: blockstack_lib::clarity_vm::clarity::ClarityInstance::begin_block
             at src/stackslib/src/clarity_vm/clarity.rs:309:17
  12: blockstack_lib::chainstate::stacks::db::StacksChainState::inner_clarity_tx_begin
             at src/stackslib/src/chainstate/stacks/db/mod.rs:2066:32
  13: blockstack_lib::chainstate::stacks::db::StacksChainState::chainstate_block_begin
             at src/stackslib/src/chainstate/stacks/db/mod.rs:1777:9
      blockstack_lib::chainstate::stacks::db::blocks::<impl blockstack_lib::chainstate::stacks::db::StacksChainState>::setup_block
             at src/stackslib/src/chainstate/stacks/db/blocks.rs:5468:30
  14: blockstack_lib::chainstate::stacks::db::blocks::<impl blockstack_lib::chainstate::stacks::db::StacksChainState>::append_block
             at src/stackslib/src/chainstate/stacks/db/blocks.rs:5837:13
  15: blockstack_lib::chainstate::stacks::db::blocks::<impl blockstack_lib::chainstate::stacks::db::StacksChainState>::process_next_staging_block
             at src/stackslib/src/chainstate/stacks/db/blocks.rs:6469:53
      blockstack_lib::chainstate::stacks::db::blocks::<impl blockstack_lib::chainstate::stacks::db::StacksChainState>::process_blocks
             at src/stackslib/src/chainstate/stacks/db/blocks.rs:6683:19
  16: blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,N,U,CE,FE,B>::process_ready_blocks
             at src/stackslib/src/chainstate/coordinator/mod.rs:2914:36
  17: blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,N,U,CE,FE,B>::inner_handle_new_burnchain_block
             at src/stackslib/src/chainstate/coordinator/mod.rs:2486:39
  18: blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,N,U,CE,FE,B>::handle_new_burnchain_block
             at src/stackslib/src/chainstate/coordinator/mod.rs:2145:9
      blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,blockstack_lib::chainstate::coordinator::comm::ArcCounterCoordinatorNotices,blockstack_lib::chainstate::coordinator::OnChainRewardSetProvider,CE,FE,B>::run
             at src/stackslib/src/chainstate/coordinator/mod.rs:424:23
  19: stacks_node::run_loop::neon::RunLoop::spawn_chains_coordinator::{{closure}}
             at src/testnet/stacks-node/src/run_loop/neon.rs:552:17
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/sys_common/backtrace.rs:135:18
  20: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/thread/mod.rs:529:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:500:40
      std::panicking::try
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:464:19
      std::panic::catch_unwind
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panic.rs:142:14
      std::thread::Builder::spawn_unchecked_::{{closure}}
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/thread/mod.rs:528:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ops/function.rs:250:5
  21: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/boxed.rs:1993:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/boxed.rs:1993:9
      std::sys::unix::thread::Thread::new::thread_start
             at rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/sys/unix/thread.rs:108:17

Dumping core for pid 1
```
This only occurs when there is existing blockchain data or you restore from a snapshot. If you don't have existing data, the node syncs.

**Steps To Reproduce**

1. Build the tip of the `develop` branch
2. Load snapshot from https://archive.hiro.so/mainnet/stacks-blockchain/ or use existing data
3. Run node

**Expected behavior**
The stacks node runs, and migrates any data sources to new schema if required. 

**Environment (please complete the following information):**
 - OS: Alpine
 - Rust version: 1.72.0
 - `develop` (https://github.com/stacks-network/stacks-blockchain/commit/2eea67546c39e54ff2bb74d8a538367615e033e6)

**Additional context**
The intent is to setup a node to do some signer / DKG testing, and thus the need to run from the tip of develop. We have a node doing a fresh sync but will take a few weeks to complete. It would be ideal if we could speed up that process by using the provided snapshots from hiro.

update: the diff/patch idea did not work. 
using a diff like this:
```diff
diff --git a/clarity/Cargo.toml b/clarity/Cargo.toml
index 523b95cba..86089991d 100644
--- a/clarity/Cargo.toml
+++ b/clarity/Cargo.toml
@@ -50,7 +50,7 @@ assert-json-diff = "1.0.0"
 # criterion = "0.3"
 
 [features]
-default = ["developer-mode"]
+default = []
 developer-mode = []
 slog_json = ["stacks_common/slog_json"]
 testing = []
diff --git a/stacks-common/Cargo.toml b/stacks-common/Cargo.toml
index 19cd58172..28c7c9e2d 100644
--- a/stacks-common/Cargo.toml
+++ b/stacks-common/Cargo.toml
@@ -62,7 +62,7 @@ rstest_reuse = "0.1.3"
 assert-json-diff = "1.0.0"
 
 [features]
-default = ["developer-mode"]
+default = []
 developer-mode = []
 slog_json = ["slog-json"]
 testing = []
diff --git a/stackslib/Cargo.toml b/stackslib/Cargo.toml
index 129ce3e8c..fcd917fd6 100644
--- a/stackslib/Cargo.toml
+++ b/stackslib/Cargo.toml
@@ -100,7 +100,7 @@ rstest = "0.17.0"
 rstest_reuse = "0.5.0"
 
 [features]
-default = ["developer-mode"]
+default = []
 profile-sqlite = []
 disable-costs = []
 developer-mode = []
 ```
 
 produces errors like this:
 ```
    Compiling clarity v0.0.1 (/home/admin/stacks-blockchain/clarity)
error[E0609]: no field `span` on type `PreSymbolicExpression`
   --> clarity/./src/vm/ast/parser/v2/mod.rs:201:29
    |
201 |                     comment.span = token.span.clone();
    |                             ^^^^ unknown field
    |
    = note: available fields are: `pre_expr`, `id`

error[E0609]: no field `span` on type `PreSymbolicExpression`
   --> clarity/./src/vm/ast/parser/v2/mod.rs:228:83
    |
228 |                         self.add_diagnostic(ParseErrors::ExpectedWhitespace, node.span.clone())?;
    |                                                                                   ^^^^ unknown field
    |
    = note: available fields are: `pre_expr`, `id`
```
    
however, removing `.clarity/Cargo.toml` from the patchfile compiles, but produces the same error reported by @EmbeddedAndroid 
 @obycode  - any ideas here?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,7 +24,7 @@ StackerDB replicas in their config files.
 
 ### Changed
 
-- `developer-mode` is not longer enabled in the default feature set. This is the correct default behavior, since the stacks-node should NOT build with developer-mode enabled by default. Tools that need to use developer-mode should enable it explicitly.
+- `developer-mode` is no longer enabled in the default feature set. This is the correct default behavior, since the stacks-node should NOT build with developer-mode enabled by default. Tools that need to use developer-mode should enable it explicitly.
 
 ### Fixed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,10 @@ StackerDB replicas in their config files.
   hosted by subscribed Stacks nodes and controlled by smart contracts
 - Added 3 new public and regionally diverse bootstrap nodes: est.stacksnodes.org, cet.stacksnodes.org, sgt.stacksnodes.org
 
+### Changed
+
+- `developer-mode` is not longer enabled in the default feature set. This is the correct default behavior, since the stacks-node should NOT build with developer-mode enabled by default. Tools that need to use developer-mode should enable it explicitly.
+
 ### Fixed
 
 - The transaction receipts for smart contract publish transactions now indicate

--- a/stackslib/Cargo.toml
+++ b/stackslib/Cargo.toml
@@ -100,7 +100,7 @@ rstest = "0.17.0"
 rstest_reuse = "0.5.0"
 
 [features]
-default = ["developer-mode"]
+default = []
 profile-sqlite = []
 disable-costs = []
 developer-mode = ["clarity/developer-mode"]

